File: Assets/Scripts/ExplosionController.js
#pragma strict

private var expireTime : float;

function Start () {
	expireTime = Time.time + 0.8;
}

function Update () {
	CheckExpiry();
}

function OnTriggerEnter (other : Collider) {
	if (other.CompareTag("Monster")) {
		other.GetComponent(MonsterController).Stun();
	}
}

function CheckExpiry () {
	if (Time.time >= expireTime) {
		Destroy(this.gameObject);
	}
}

File: Assets/Scripts/ExplosionController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ExplosionController : MonoBehaviour
{
    private float expireTime;

    void Start()
    {
        expireTime = (Time.time + 0.8f);
    }

    void Update()
    {
        CheckExpiry();
    }

    void OnTriggerEnter(Collider other)
    {
        if (other.CompareTag("Monster"))
        {
            other.GetComponent<MonsterController>().Stun();
        }
    }

    void CheckExpiry()
    {
        if (Time.time >= expireTime)
        {
            Destroy(this.gameObject);
        }
    }
}
